Source/WebCore/editing/markup.h

```diff
@@ -32,6 +32,7 @@
 #include "HTMLInterchange.h"
 #include "MarkupExclusionRule.h"
 #include "ParserContentPolicy.h"
+#include "ShadowRoot.h"
```

Comment:
It seems like this should just be `class ShadowRoot`, I agree, but
```
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/DerivedSources/WebCore/unified-sources/UnifiedSource443.cpp:1:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/WebCorePrefix.h:165:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/HashMap.h:25:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/HashTable.h:33:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/HashTraits.h:26:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/HashFunctions.h:26:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/RefPtr.h:28:
/Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/Ref.h:62:16: error: member access into incomplete type 'WebCore::ShadowRoot'
   62 |             ptr->deref();
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/DerivedSources/WebCore/unified-sources/UnifiedSource443.cpp:1:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/WebCorePrefix.h:165:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/HashMap.h:25:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/HashTable.h:33:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/HashTraits.h:26:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/HashFunctions.h:26:
/Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/RefPtr.h:60:47: note: in instantiation of member function 'WTF::DefaultRefDerefTraits<WebCore::ShadowRoot>::derefIfNotNull' requested here
   60 |     ALWAYS_INLINE ~RefPtr() { RefDerefTraits::derefIfNotNull(PtrTraits::exchange(m_ptr, nullptr)); }
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/DerivedSources/WebCore/unified-sources/UnifiedSource443.cpp:1:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/WebCorePrefix.h:168:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/text/AtomString.h:24:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/text/AtomStringImpl.h:23:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/text/UniquedStringImpl.h:28:
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/text/StringImpl.h:35:
/Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/Vector.h:70:19: note: in instantiation of member function 'WTF::RefPtr<WebCore::ShadowRoot>::~RefPtr' requested here
   70 |             cur->~T();
/Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/Vector.h:253:73: note: in instantiation of member function 'WTF::VectorDestructor<true, WTF::RefPtr<WebCore::ShadowRoot>>::destruct' requested here
  253 |         VectorDestructor<!std::is_trivially_destructible<T>::value, T>::destruct(begin, end);
/Users/annevk/Dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/Vector.h:781:29: note: in instantiation of member function 'WTF::VectorTypeOperations<WTF::RefPtr<WebCore::ShadowRoot>>::destruct' requested here
  781 |             TypeOperations::destruct(begin(), end());
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/DerivedSources/WebCore/unified-sources/UnifiedSource443.cpp:5:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/xml/XSLTProcessor.cpp:41:
/Users/annevk/Dev/OpenSource/Source/WebCore/editing/markup.h:106:426: note: in instantiation of member function 'WTF::Vector<WTF::RefPtr<WebCore::ShadowRoot>>::~Vector' requested here
  106 | WEBCORE_EXPORT String serializeFragment(const Node&, SerializedNodes, Vector<Ref<Node>>* = nullptr, ResolveURLs = ResolveURLs::No, std::optional<SerializationSyntax> = std::nullopt, HashMap<String, String>&& replacementURLStrings = { }, HashMap<RefPtr<CSSStyleSheet>, String>&& replacementURLStringsForCSSStyleSheet = { }, IncludeShadowRoots = IncludeShadowRoots::ExplicitOnly, Vector<RefPtr<ShadowRoot>> explicitShadowRoots = { }, const Vector<MarkupExclusionRule>& exclusionRules = { });
In file included from /Users/annevk/Dev/OpenSource/WebKitBuild/Release/DerivedSources/WebCore/unified-sources/UnifiedSource443.cpp:1:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/xml/XPathVariableReference.cpp:28:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/xml/XPathVariableReference.h:28:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/xml/XPathExpressionNode.h:29:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/xml/XPathValue.h:29:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/xml/XPathNodeSet.h:29:
In file included from /Users/annevk/Dev/OpenSource/Source/WebCore/dom/Node.h:33:
/Users/annevk/Dev/OpenSource/Source/WebCore/dom/TreeScope.h:59:7: note: forward declaration of 'WebCore::ShadowRoot'
   59 | class ShadowRoot;
1 error generated.
```
and `CSSStyleSheet` follows the same pattern.

Reply:
Yeah, ~Vector<RefPtr<ShadowRoot>> will need to see the definition of ShadowRoot.